src/hex/vertex.rs

```diff
@@ -164,14 +186,88 @@ impl Vertex {
             ]
         }
     }
+
+    /// Generate the edges adjacent to this vertex.
+    ///
+    /// ```
+    /// use gridava::hex::vertex::{Vertex, VertexSpin, vertex};
+    ///
+    /// let edges = vertex!(0,0,VertexSpin::Up).adjacent_edges();
+    /// ```
+    pub fn adjacent_edges(&self) -> [Edge; 3] {
+        match self.spin {
+            VertexSpin::Up => [
+                edge!(self.q + 1, self.r - 1, EdgeDirection::West),
+                edge!(self.q, self.r, EdgeDirection::NorthEast),
+                edge!(self.q, self.r, EdgeDirection::NorthWest),
+            ],
+            VertexSpin::Down => [
+                edge!(self.q, self.r + 1, EdgeDirection::NorthWest),
+                edge!(self.q, self.r + 1, EdgeDirection::West),
+                edge!(self.q - 1, self.r + 1, EdgeDirection::NorthEast),
+            ],
+        }
+    }
+
+    /// Compute the L1 distance between two vertices.
+    ///
+    /// ```
+    /// use gridava::hex::vertex::{Vertex, VertexSpin, vertex};
+    ///
+    /// let dist = vertex!(0,0,VertexSpin::Up).distance(vertex!(1,0,VertexSpin::Up));
+    /// ```
+    pub fn distance(&self, b: Self) -> i32 {
```

Comment:
This is a really excellent algorithm.

Reply:
Took me forever to figure it out, the conditional inside the function body is the only one in the path as well.